### PR TITLE
New Map extension

### PR DESCRIPTION
**Pull request by JanSchankin**
+ Maps with primitve wrapper types as keys can now be persisted

### CODE REVIEW

File: deepsampler-persistence/src/main/java/de/ppi/deepsampler/persistence/bean/ext/MapPrimitiveKeyExtension.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+        ((Map<String, Object>) persistentBean).entrySet().stream()
+                .map(entry -> {
+                    Object key = ReflectionTools.parseString(entry.getKey(), keyClass);
+                    Object value = persistentBeanConverter.revert(entry.getValue(), valueClass, valueType);
+                    return new Object[]{key, value};
+                })
+                .forEach(pair -> valueMap.put(pair[0], pair[1]));
+
+        return (T) valueMap;
```
Comment (rcschrg) [inline deepsampler-persistence/src/main/java/de/ppi/deepsampler/persistence/bean/ext/MapPrimitiveKeyExtension.java:129]: I think a conventional for each would be more readable, multi-line lambda + entryMap stream is too much in my opinion.

File: deepsampler-persistence/src/main/java/de/ppi/deepsampler/persistence/bean/ext/MapPrimitiveKeyExtension.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        ((Map<Object, Object>) originalBean).entrySet().stream()
+                .map(entry -> {
+                    Object value = persistentBeanConverter.convert(entry.getValue(), entryType);
+                    String key = entry.getKey() != null ? entry.getKey().toString() : "null";
+
+                    return new Object[]{key, value};
+                })
+                .forEach(pair -> convertedMap.put(pair[0], pair[1]));
+
```
Comment (rcschrg) [inline deepsampler-persistence/src/main/java/de/ppi/deepsampler/persistence/bean/ext/MapPrimitiveKeyExtension.java:67]: same as below

**Review by rcschrg — APPROVED** [review]
I like the change of the API for the extensions, but we might want some simpler APIs for common use cases in the future, but thats another issue I guess.

**Comment by JanSchankin** [comment]
> I like the change of the API for the extensions, but we might want some simpler APIs for common use cases in the future, but thats another issue I guess.

I've created an issue for that (#69).

**Pull request merged**